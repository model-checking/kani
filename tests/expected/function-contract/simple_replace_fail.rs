// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts -Zstubbing

#[kani::requires(divisor != 0)]
#[kani::ensures(|result : &u32| *result <= dividend)]
fn div(dividend: u32, divisor: u32) -> u32 {
    dividend / divisor
}

#[kani::proof]
#[kani::stub_verified(div)]
fn main() {
    assert!(div(9, 4) == 2, "contract doesn't guarantee equality");
}

#[kani::proof_for_contract(div)]
fn check_div() {
    let dividend: u32 = kani::any();
    let divisor: u32 = kani::any();
    div(dividend, divisor);
}
