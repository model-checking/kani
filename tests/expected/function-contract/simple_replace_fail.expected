main.assertion\
- Status: FAILURE\
- Description: ""contract doesn't guarantee equality""
